### PR TITLE
FR-9201 - Proxy middlewares removal and general enhancements

### PR DESCRIPTION
**Pull request by eran-frontegg**

### CODE REVIEW

File: .github/workflows/_build-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build the lib
+        run: npm run build --if-present
```
Comment (raz-shlomo-frontegg) [inline .github/workflows/_build-and-test.yaml:25]: why --if-present?

File: .github/workflows/on-pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: On Pull Request
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+      - converted_to_draft
```
Comment (raz-shlomo-frontegg) [inline .github/workflows/on-pull-request.yaml:10]: why should it run in case of draft?

File: .npmignore [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+
+.fleet
+.idea
+
+node_modules
+
+.eslintignore
+.eslintrc.json
+.prettierignore
+.prettierrc
```
Comment (raz-shlomo-frontegg) [inline .npmignore:19]: Let's talk about this file please

File: package.json [hunk 4]
```diff
@@ -5,10 +5,11 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "tsc",
+    "build": "rm -rf dist && tsc",
     "lint": "eslint --ignore-path .eslintignore --ext .js,.ts .",
     "format": "prettier --write \"**/*.+(js|ts|json)\"",
     "test": "npm run build && jest",
+    "test:coverage": "npm test -- --coverage",
```
Comment (raz-shlomo-frontegg) [inline package.json:12]: coverage threshold block is missing

**Review by raz-shlomo-frontegg — APPROVED** [review]

**Review by aviadmizrachi — APPROVED** [review]

**Pull request merged**